planner/languages/java/java_planner.go

```diff
@@ -83,62 +99,134 @@ func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
 	}
 }
 
+func (p *Planner) packageManager(srcDir string) (string, error) {
+	pomXMLPath := filepath.Join(srcDir, mavenFileName)
+	buildGradlePath := filepath.Join(srcDir, gradleFileName)
+	if plansdk.FileExists(pomXMLPath) {
+		return MavenType, nil
+	} else if plansdk.FileExists(buildGradlePath) {
+		return GradleType, nil
+	} else {
+		return "", errors.New("Could not locate a Maven or Gradle file.")
+	}
+}
+
+func (p *Planner) devPackages(srcDir string, builderTool string) ([]string, error) {
+	javaPkg, err := getJavaPackage(srcDir, builderTool)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	devPackagesMap := map[string][]string{
+		MavenType: {
+			defaultMaven,
+			javaPkg,
+			binUtils,
+		},
+		GradleType: {
+			defaultGradle,
+			javaPkg,
+			binUtils,
+		},
+	}
+
+	return devPackagesMap[builderTool], nil
+}
+
+func (p *Planner) runtimePackages(builderTool string) []string {
+	return []string{
+		binUtils,
+	}
+}
+
 // This method is added because we plan to differentiate Gradle and Maven.
 // Otherwise, we could just assign the value without calling this.
-func (p *Planner) installCommand(srcDir string) string {
-	// TODO: Add support for Gradle install command
-	return "mvn clean install"
+func (p *Planner) installCommand(builderTool string) string {
+	installCommandMap := map[string]string{
+		MavenType:  "mvn clean install",
+		GradleType: "./gradlew build",
+	}
+	return installCommandMap[builderTool]
 }
 
-func (p *Planner) buildCommand() string {
-	return "jlink --verbose " +
-		"--add-modules ALL-MODULE-PATH " +
-		"--strip-debug" +
-		" --no-man-pages " +
-		" --no-header-files " +
-		" --compress=2 " +
-		"--output ./customjre"
+func (p *Planner) buildCommand(builderTool string) string {
+	return "jlink --verbose" +
+		" --add-modules ALL-MODULE-PATH" +
+		" --strip-debug" +
+		" --no-man-pages" +
+		" --no-header-files" +
+		" --compress=2" +
+		" --output ./customjre"
 }
 
-func (p *Planner) startCommand(srcDir string) (string, error) {
-	pomXMLPath := fmt.Sprintf("%s/pom.xml", srcDir)
-	var parsedPom mvnparser.MavenProject
-	err := cuecfg.ParseFile(pomXMLPath, &parsedPom)
-	if err != nil {
-		return "", errors.WithMessage(err, "error parsing the pom file")
+func (p *Planner) startCommand(srcDir string, builderTool string) (string, error) {
+	if builderTool == MavenType {
+		pomXMLPath := fmt.Sprintf("%s/%s", srcDir, mavenFileName)
+		var parsedPom mvnparser.MavenProject
+		err := cuecfg.ParseFile(pomXMLPath, &parsedPom)
+		if err != nil {
+			return "", errors.WithMessage(err, "error parsing the pom file")
+		}
+		return fmt.Sprintf("./customjre/bin/java -jar target/%s-%s.jar", parsedPom.ArtifactId, parsedPom.Version), nil
+	} else if builderTool == GradleType {
+		return "export JAVA_HOME=./customjre && ./gradlew run", nil
```

Comment:
could this be simplified to:
```
JAVA_HOME=./customjre ./gradlew run
```
?

Reply:
without export it doesn't seem to work. So I'll just leave it there.